File: .pub-cache/hosted/pub.dev/zego_express_engine-3.23.0/android/src/main/java/im/zego/zego_express_engine/ZegoCustomEventManager.java
//
//  ZegoCustomEventManager.java
//  im.zego.zego_express_engine
//
//  Created by zego on 2025/10/31.
//

package im.zego.zego_express_engine;

public class ZegoCustomEventManager {
    private volatile static ZegoCustomEventManager instance;
    private IZegoFlutterCustomEventHandler handler;

    private ZegoCustomEventManager() {
    }

    /**
     * Get the custom event manager instance
     */
    public static synchronized ZegoCustomEventManager getInstance() {
        if (instance == null) {
            synchronized (ZegoCustomEventManager.class) {
                if (instance == null) {
                    instance = new ZegoCustomEventManager();
                }
            }
        }
        return instance;
    }

    /**
     * Set custom event handler
     */
    public void setCustomEventHandler(IZegoFlutterCustomEventHandler handler) {
        this.handler = handler;
    }

    /**
     * Get custom event handler
     */
    public IZegoFlutterCustomEventHandler getCustomEventHandler() {
        return handler;
    }
}

File: .pub-cache/hosted/pub.dev/zego_express_engine-3.23.0/windows/ZegoCustomEventManager.cpp
#include "include/zego_express_engine/ZegoCustomEventManager.h"

static std::shared_ptr<ZegoCustomEventManager> instance_ = nullptr;
static std::once_flag singletonFlag;

std::shared_ptr<ZegoCustomEventManager> ZegoCustomEventManager::getInstance() {
    std::call_once(singletonFlag, [&] {
        instance_ = std::make_shared<ZegoCustomEventManager>();
    });
    return instance_;
}

std::shared_ptr<IZegoFlutterCustomEventHandler> ZegoCustomEventManager::getHandler() {
    return handler_;
}

void ZegoCustomEventManager::setCustomEventHandler(std::shared_ptr<IZegoFlutterCustomEventHandler> handler) {
    handler_ = handler;
}   